src/e2e-test/java/io/cdap/plugin/gcscopy/stepsdesign/GCSCopy.java

```diff
@@ -0,0 +1,158 @@
+  }
+
+  @Then("Enter GCSCopy property destination path {string}")
+  public void enterGCSCopyPropertyDestinationPath(String path) {
+    GCSCopyActions.enterDestinationPath("gs://" + TestSetupHooks.gcsTargetBucketName
+                                          + "/" + PluginPropertyUtils.pluginProp(path));
+  }
+
+  @Then("Validate GCSCopy successfully copied object {string} to destination bucket in location {string}")
+  public void validateGCSCopySuccessfullyCopiedObjectToDestinationBucketInLocation(String path, String location) {
+    String targetGCSBucket = TestSetupHooks.gcsTargetBucketName;
+    try {
+      if (StorageClient.getBucketMetadata(targetGCSBucket).getLocation().equalsIgnoreCase(PluginPropertyUtils.pluginProp
+        (location))) {
+        validateGCSCopySuccessfullyCopiedObjectToDestinationBucket(path);
+        return;
+      }
+      Assert.fail("Target gcs bucket " + targetGCSBucket + " is not created in location " + location);
+    } catch (StorageException | IOException e) {
+      if (e.getMessage().contains("The specified bucket does not exist")) {
+        Assert.fail("Target gcs bucket " + targetGCSBucket + " not created - " + e.getMessage());
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  @Then("Enter runtime argument value {string} for GCSCopy property sourcePath key {string}")
+  public void enterRuntimeArgumentValueForGCSCopyPropertySourcePathKey(String value, String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey),
+                           "gs://" + TestSetupHooks.gcsSourceBucketName + "/" + PluginPropertyUtils.pluginProp(value));
+  }
+
+  @Then("Enter runtime argument value for GCSCopy property destination path key {string}")
+  public void enterRuntimeArgumentValueForGCSCopyPropertyDestinationPathKey(String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey),
+                           "gs://" + TestSetupHooks.gcsTargetBucketName);
+  }
+
+  @Then("Validate GCSCopy did not copy subdirectory {string} to destination bucket")
+  public void validateGCSCopyDidNotCopySubdirectoryToDestinationBucket(String subdirectory) throws IOException {
+    String sourceGCSBucket = TestSetupHooks.gcsSourceBucketName;
+    String gcsObject = PluginPropertyUtils.pluginProp(subdirectory);
+    for (Blob blob : StorageClient.listObjects(sourceGCSBucket).iterateAll()) {
+      if (blob.getName().contains(gcsObject)) {
+
+        BeforeActions.scenario.write("Subdirectory is not copied from source bucket" + sourceGCSBucket);
+        return;
+      }
+    }
+      Assert.fail("Subdirectory is copied from source GCS Bucket " + sourceGCSBucket);
+    }
+
+  @Then("Validate GCSCopy failed to copy object {string} to destination bucket")
+  public void validateGCSCopyFailedToCopyObjectToDestinationBucket(String path) throws IOException {
```

Comment:
this only seems to check the existence of the object in the source bucket. Shouldn't it also validate the non-existence of the object in the destination bucket ?

Reply:
So here this step is used to check that the pipeline is getting failed when we are selecting overwrite existing file as false. Here we already have the file it the destination bucket and we are selecting overwrite option as false and the pipeline is failing.So here we already have the file in the destination bucket.